soh/soh/Enhancements/mods.cpp

```diff
@@ -258,6 +256,52 @@ void RegisterRupeeDash() {
     });
 }
 
+void RegisterBonkDamage() {
+    GameInteractor::Instance->RegisterGameHook<GameInteractor::OnPlayerBonk>([]() {
+        uint8_t bonkOption = CVarGetInteger("gBonkDamageMul", 0);
+        uint16_t bonkDamage = 0;
+        switch (bonkOption) {
+            // Quarter heart
+            case 1:
+                bonkDamage = 4;
+                break;
+            // Half a heart
+            case 2:
+                bonkDamage = 8;
+                break;
+            // Full heart
+            case 3:
+                bonkDamage = 16;
+                break;
+            // 2 hearts
+            case 4:
+                bonkDamage = 32;
+                break;
+            // 4 hearts
+            case 5:
+                bonkDamage = 64;
+                break;
+            // 8 hearts
+            case 6:
+                bonkDamage = 128;
+                break;
+            case 0:
+            case 7:
+            default:
+                break;
+        }
+        // OHKO
+        if (bonkOption == 7) {
+            gSaveContext.health = 0;
+        } else if (bonkDamage) {
+            Health_ChangeBy(gPlayState, -bonkDamage);
```

Comment:
The raw action is set up to take full hearts as an argument and also runs a function that plays the little "hurt" animation which cancels the bonk animation if used here. Figured it wasn't a big deal since it's only 3 lines of code anyway.